### ISSUE TITLE
sorting doesn't work

### ISSUE THREAD

**Issue by gaydenko** [escherdragon/sunrise-commander#35]
Hi! At my case sorting doesn't work: hitting `s` with something else just keep sorting by name (adding `S` or `t` after date-time field for size and time qualifiers respectively). SR-related custom variables are:

```el
 '(sr-attributes-display-mask (quote (nil nil nil nil t t t)))
 '(sr-avfs-root "/home/user/.avfs")
 '(sr-listing-switches
   "--dired --group-directories-first -alh --time-style=+%y-%m-%d %H:%M:%S")
 '(sr-mirror-unionfs-impl (quote unionfs-fuse))
 '(sr-show-hidden-files t)
```
Without custom variables loading sorting does work.

**Comment by escherdragon** [comment]
Hello Andrew,

There was indeed an error in the way the sorting switches were handled to
ls. I've just committed c559886206 to master, which fixes that problem.
Thanks for reporting.

Unfortunately while investigating this I found one more problem with the
workaround I suggested previously for including "spaces" (unicode blanks,
actually) in the value of the time-style ls option -- it turns out that the
unicode characters do not match the
`directory-listing-before-filename-regexp' regular expression, which is
used by dired for implementing several important operations (e.g. finding
different places in the buffer). This is why even with the fix for #35
you'll encounter other problems with sorting (e.g. it won't work in virtual
panes).

If displaying the seconds in the date of your entries in the pane is not
crucial for you, then I'd recommend using --time-style=long-iso instead.
Another possibility would be to write a small wrapper around ls that passes
the exact options required by you to ls and use that instead as your
`insert-directory-program', at least as long as the bug in dired
<http://debbugs.gnu.org/cgi/bugreport.cgi?bug=18875> is not yet fixed.

Cheers,

On Wed, Jun 17, 2015 at 8:45 PM, Andrew Gaydenko <notifications@github.com>
wrote:

> Hi! At my case sorting doesn't work: hitting s with something else just
> keep sorting by name (adding S or t after date-time field for size and
> time qualifiers respectively). SR-related custom variables are:
>
>  '(sr-attributes-display-mask (quote (nil nil nil nil t t t)))
>  '(sr-avfs-root "/home/user/.avfs")
>  '(sr-listing-switches
>    "--dired --group-directories-first -alh --time-style=+%y-%m-%d %H:%M:%S")
>  '(sr-mirror-unionfs-impl (quote unionfs-fuse))
>  '(sr-show-hidden-files t)
>
> Without custom variables loading sorting does work.
>

-- 
José A. Romero L.
escherdragon@gmail.com
"We who cut mere stones must always be envisioning cathedrals."
(Quarry worker's creed)
